docs/source/introduction.md

```diff
@@ -181,6 +182,63 @@ with the bbox.
 
 Note: the `bbox` values must be in the same CRS as the dataset.
 
+## Execute a sql query
+
+You can use the `sql` parameter to execute a sql query on a dataset. 
+
+Depending on the dataset, you can use different sql dialects. By default, if 
+the dataset natively supports sql, the sql statement will be passed through 
+as such. Hence, the sql query should be written in the relevant native sql
+dialect (e.g. [GeoPackage](https://gdal.org/drivers/vector/gpkg.html)/
+[Sqlite](https://gdal.org/drivers/vector/sqlite.html), 
+[PostgreSQL](https://gdal.org/drivers/vector/pg.html)). If the datasource 
+doesn't natively support sql (e.g. 
+[ESRI Shapefile](https://gdal.org/drivers/vector/shapefile.html), 
+[FlatGeobuf](https://gdal.org/drivers/vector/flatgeobuf.html)), you can choose 
+between '[OGRSQL](https://gdal.org/user/ogr_sql_dialect.html#ogr-sql-dialect)' 
+(the default) and  
+'[SQLITE](https://gdal.org/user/sql_sqlite_dialect.html#sql-sqlite-dialect)'. 
+For SELECT statements the 'SQLITE' dialect tends to provide more spatial 
+features as all 
+[spatialite](https://www.gaia-gis.it/gaia-sins/spatialite-sql-latest.html) 
+functions can be used. 
```

Comment:
Seems like given your experience with the CI not supporting this, it would be good to note that your GDAL needs to have spatialite support built in, as it is an optional component?

Reply:
This is also specified in the GDAL documentation, but I'll mention it here as well.